### ISSUE TITLE
[Ideal Nav] [$1000] [Tracking] Update navigation to support wide layouts on native

### ISSUE THREAD

**Issue by roryabraham** [Expensify/App#35854]
Slack context: https://expensify.slack.com/archives/C07J32337/p1707147391183469

### Problem
We have an iPad app, and it doesn't work as expected, because we have code that assumes that any code in `index.native.js` _must_ be representing a narrow layout. Using _platform_ as a proxy for:

- device layout
- presence of a mouse/hoverability
- keyboard handling

is a bad practice and a source for technical debt. It also violates one of the [driving philosophies of this app](https://github.com/Expensify/App?tab=readme-ov-file#Philosophy) - cross platform 99.99%.

For the sake of scope management, this issue will focus on removing instances where platform is used as a proxy for device layout, specifically these known issues, which will soon be fixed via a [temporary workaround](https://github.com/Expensify/App/pull/35845).

- https://github.com/Expensify/App/issues/35753
- https://github.com/Expensify/App/issues/35755
- https://github.com/Expensify/App/issues/35752
- https://github.com/Expensify/App/issues/35758
- https://github.com/Expensify/App/issues/35759
- https://github.com/Expensify/App/issues/35764

### Solution
Remove code in `index.native.js` files that assumes it will be running on a narrow layout. Let's get wide layouts working on an iPad Pro (in portrait mode). When [this was investigated before](https://github.com/Expensify/App/issues/26421#issuecomment-1707975034), one of the key difficulties found was that we rely on `position: fixed` in wide layouts on web, but [that style is not currently supported in React Native](https://github.com/facebook/yoga/issues/264).

This does not mean we will allow the native apps to rotate into a landscape mode. 
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01290d6331a572c985</li>
        <li>Upwork Job ID: 1754814923483574272</li>
        <li>Last Price Increase: 2024-08-08</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @chuckdries</details>

**Comment by roryabraham** [comment]
cc @mountiny @hayata-suenaga @adamgrzybowski @WojtekBoman @mateuuszzzzz @kosmydel @MaciejSWM

**Comment by roryabraham** [comment]
Maybe one alternative to using `position: fixed` would be to leverage `@gorhom/portal` to move the view outside the positional scope, as suggested [here](https://github.com/facebook/yoga/issues/264#issuecomment-354728960)

**Comment by hayata-suenaga** [comment]
What if we create a parent component that fills the entire screen with its position set to `relative`, and then make the target component a child of this parent component, setting its position to `absolute`?

**Comment by mountiny** [comment]
Noted in the OP we still do not want to allow the landscape mode. Making this External to get proposals going. If anything is not clear, please ask. Thanks

**Comment by hayata-suenaga** [comment]
Can I close my issue [Unable to navigate away from Profile page using the back button](https://github.com/Expensify/App/issues/35753) (which is already linked on the OP) as the issue will be fixed in this issue?

**Comment by roryabraham** [comment]
[From Nick Gerleman at Meta](https://discord.com/channels/514829729862516747/1113127476424622110/1204598960577314867):

> We actually just spent a couple months adding support for position: "static" in the new architecture.
> 
> I will warn that adding position: "fixed" would be fairly complicated, and break a good number of assumptions in both Yoga and Fabric. E.g. beyond layout, it we would need to change Fabric mounting layer as well (e.g. we couldn't put a fixed item under a scrolling container. IIRC fixed has stacking context implications as well).
> 
> I would discourage trying to take it on without previous experience in both Yoga and Fabric's mounting later.

**Comment by adamgrzybowski** [comment]
The position fixed is used because we need to modify the position of cards inside the StackView component provided by react-navigation, to display them side by side.

As a potential idea to investigate, we could try to use two or more StackViews in the custom navigator. This should give us more flexibility with positioning. 

Alternatively, we could create our own / modify the StackView.

However, I am not sure how it would work with native stacks

**Comment by mountiny** [comment]
Bumping this to hopefully get some proposals here

**Comment by sakluger** [comment]
Posted in the Callstack Slack room to ask for volunteers.

**Comment by rezkiy37** [comment]
Hi, I’m Michael (Mykhailo) from Callstack and I would like to work on this issue.

**Comment by badeggg** [comment]
Does this issue still need proposal?

**Comment by rezkiy37** [comment]
I am working to prepare a proposal.

**Comment by rezkiy37** [comment]
Actively working on the issue. Resolving a problem with 2 active stacks on a screen at the same moment.

**Comment by rezkiy37** [comment]
Still trying to implement 2 active stacks on a screen at the same moment.

**Comment by rezkiy37** [comment]
Discuss internally how to handle the feature with 2 active stacks on a screen at the same moment.

**Comment by rezkiy37** [comment]
I need to create a custom navigator to handle this case on native devices (_iPads_). So, I am going to develop it. Btw, looks like it is a hard task, therefore I need some time to investigate and implement.

**Comment by trjExpensify** [comment]
👋 can someone catch me up on why this one was added to #wave8, and should it now be transfered to #wave-collect?

**Comment by rezkiy37** [comment]
Going to integrate something similar to what they described in this [article](https://dev.to/retyui/how-to-implement-split-view-on-a-tablet-with-react-native-167e). Testing on a bare project now.

**Comment by mountiny** [comment]
@rezkiy37 Just wanted point you also towards this ideal nav PR which makes some updates to the navigators https://github.com/Expensify/App/pull/37421

@trjExpensify This is a follow up to the ideal nav / app navigation reboot updates, we want to clean up how the layout is handled based on the device size. 

At the moment, some actions are decided based on the screen width which because on native devices we struggled to achieve the desktop layout due to some RN styling limitations. We want to remove this debt. 

It was added as a follow up to ideal nav then

**Comment by trjExpensify** [comment]
Okay, so @mountiny I'm putting this one in #wave-collect polish. I've also added `[Ideal Nav]` in the issue title to associate it with that project.

**Comment by mountiny** [comment]
Thank you!

**Comment by rezkiy37** [comment]
I have some progress on the issue. The idea is to have 2 `NavigationContainers`. Continue working on it.


<img width="726" alt="Screenshot 2024-03-12 at 18 47 22" src="https://github.com/Expensify/App/assets/57314631/9151517f-1da5-42aa-9576-23b6e953b47a">
<img width="1728" alt="Screenshot 2024-03-12 at 18 47 27" src="https://github.com/Expensify/App/assets/57314631/883d9e53-0883-45b1-9e70-e9306321b1d6">

**Comment by rezkiy37** [comment]
Hey!
I must inform you that I have a vacation next week. I will be back on Monday 25.03. During the vacation, someone from Callstack can take the issue to work. See you soon!

**Comment by trjExpensify** [comment]
Welcome back, @rezkiy37! Will you be able to finish this off this week?

**Comment by rezkiy37** [comment]
Hello!
Yeah, I am going to work on the issue actively. Also, I have the high-priority Invoicing V1 feature. I will post daily updates so will see. Thank you!

**Comment by rezkiy37** [comment]
I have some progress today. The app has started to render 2 stacks. Also, I was able to integrate a draft communication between them. 

https://github.com/Expensify/App/assets/57314631/43a33c5c-2a6e-4a58-ba53-70a65394ae12


https://github.com/Expensify/App/assets/57314631/7432c3bb-5114-4a62-8a65-ec5c6b0741c9

**Comment by mountiny** [comment]
Nice thank you for the update

**Comment by rezkiy37** [comment]
Another high-priority task occupied me. I am going to focus on this one tomorrow.

**Comment by rezkiy37** [comment]
**Updates:** I was working on a behavior to persist previous pages for the tabs. The app remembers a chat and settings pages to get the proper previous page to open via the tabs. Also, I was able to configure the workspace settings page. The next goal is to configure all modal stacks to be workable.

https://github.com/Expensify/App/assets/57314631/205a2da1-0f0f-49c7-b7dd-6a6fdc13e50e

**Comment by rezkiy37** [comment]
**Updates:** I've configured the modal stacks to open and work with them. The next goal is to have the overlay effect.


https://github.com/Expensify/App/assets/57314631/c465b9f2-c943-46a1-9941-90f9781c7e6f

**Comment by rezkiy37** [comment]
I am stopping working on the issue as it was discussed in [Slack](https://callstack-hq.slack.com/archives/C036QM0SLJK/p1712671490554669?thread_ts=1712620008.643799&cid=C036QM0SLJK). So I am leaving all my investigation in this comment.

I have a [draft PR](https://github.com/Expensify/App/pull/37993) with a raw implementation. Unfortunately, I didn't find a solution to integrate 2 navigators to work in parallel within one **NavigationContainer**. The main idea was to create an independent **NavigationContainer** for the left side of a screen (_bottom tab navigator_). It allows us to place them near each other. Also, I needed to create one more navigation handler because the bottom tab navigator is no longer part of the **RootStack**.  

I pushed all changes to the PR. Feel free to ask any question 🙂

**Comment by allroundexperts** [comment]
As next steps, I'll be taking a first peek at this tomorrow. I think I have a different way in my mind than the draft PR, but will experiment with it tomorrow.

**Comment by mountiny** [comment]
Ok, thanks, please keep us updated here, I would prefer to not to spend too much time in the design phase here, unless you have some clear reasons why the current approach in the PR from @rezkiy37 is wrong

**Comment by mountiny** [comment]
@allroundexperts What is the latest? I think you are working on some other projects too, would you like to hand this one off to someone else?

**Comment by allroundexperts** [comment]
I'd be occupied with a bug on actionable whisper this week. If this is urgent, please re-assign.

**Comment by mountiny** [comment]
Asked in C+ channel for new volunteeer

**Comment by mountiny** [comment]
Discussed in Slack @mananjadhav will take over. Can you please make a steady progress on this with frequent updates? Feel free to ask Michael for any help too

**Comment by mananjadhav** [comment]
I'll check this over the weekend and start with this Monday.

**Comment by adamgrzybowski** [comment]
Hi guys! This issue is not easy, so I want to share some of my finding. Maybe this will help in some way!

I see a lot of progress with the above approach, but I was wondering if you had considered using two `StackViews`? Especially if communication between two `NavigationContainers` proves difficult to implement.

I tried putting two `StackViews` in the render function for a custom `StackNavigator`.

One thing that seems to be beneficial is that we only change the way the screens are displayed. I think this may require less modification to the logic since we are only using one `NavigationContainer`.

The disadvantage is that on a small screen we would have to create the transition animation from the sidebar to the central panel screens ourselves. This shouldn't be too difficult though.

Alternatively, we could consider implementing our `SplitStackView` based on the original `StackView`, but with the ability to display screens side by side

I am including very early POC of this idea below:

cc: @rezkiy37 @mananjadhav 
<img width="639" alt="image" src="https://github.com/Expensify/App/assets/67908363/65a54dce-1c20-44de-bec7-9f76c54a8b04">
<img width="1009" alt="image" src="https://github.com/Expensify/App/assets/67908363/5bf2bc27-5562-4952-b9e6-b401fc773220">

**Comment by sakluger** [comment]
@greg-schroeder assigning you to help watch this issue while I'm OOO through May 31. Please handle payment if needed. I'll take back over if it's still open when I return.

**Comment by mananjadhav** [comment]
Thanks @adamgrzybowski I was exploring SplitView instead of rewriting the NavigationContainer, but I didn't make any progress.  I am not sure if we are supporting orientation at the moment, but do you think it could cause any issues there? 

Also do you think handling the back actions/navigations would be challenging if we have nested navigators such as Tab navigator in one the StackViews  and Stack navigator in the other one?

This definitely sounds like a better plan than going for independent NavigationContainers. Lastly do you think if this will add any overhead in terms of performance/memory?

**Comment by adamgrzybowski** [comment]
@mananjadhav 
> I am not sure if we are supporting orientation at the moment, but do you think it could cause any issues there?

We don't support it yet but I guess it should be supported if we manage to support wide layouts on native. 

Switching between narrow and wide is potentially problematic, but we have it on the web, and it works in most cases. I think it is important to handle switching the same way on the web and native.

> Also do you think handling the back actions/navigations would be challenging if we have nested navigators such as Tab navigator in one the StackViews and Stack navigator in the other one?

> This definitely sounds like a better plan than going for independent NavigationContainers. Lastly do you think if this will add any overhead in terms of performance/memory?

So currently we achieve wide layout on the web with styles that move the first `StackView` card a little left and therefore making it visible. The first card is the `BottomTabNavigator` and includes every left pane screen.

If we decide to use multiple `StackView`* components with similar strategy which is to move first rendered route a little left, then I don't think it will make any significant difference in terms of navigation logic and performance. 

Also, there is one thing worth keeping in mind. I saw @JmillsExpensify mentioned about upcoming three pane view. It would be great to get more details about the new requirements. There is a possibility that if we want to have three pane view then the approach with two navigation containers will show some advantages that we are not aware now.

*We can also think about creating our own `SplitView` based on the react-navigation `StackView` but the idea is similar.

cc: @WojtekBoman @Kicu for visibility

**Comment by mountiny** [comment]
> We don't support it yet but I guess it should be supported if we manage to support wide layouts on native.

Yeah I would think we should build this with landscape orientation in mind

> There is a possibility that if we want to have three pane view then the approach with two navigation containers will show some advantages that we are not aware no

Yep that is a good point. I think that we would like to explore this option (its similar to how slack has LHN, main report and a thread in RHP), although I am not sure if right now is the best time yet.

**Comment by mountiny** [comment]
We are most likely not going to handle the three pane view until after August

**Comment by mananjadhav** [comment]
Good discussion, thanks for clarifying the details. Working on the PR. Should have something in a day or two.

**Comment by greg-schroeder** [comment]
Thanks @mananjadhav!

**Comment by mountiny** [comment]
@mananjadhav how is it looking

**Comment by mananjadhav** [comment]
I tried working on both the approaches and I had some challenges. But @adamgrzybowski's approach is more promising so that's what I'll work on now. I'll have more updates by the weekend.

**Comment by greg-schroeder** [comment]
Appreciate the update!

**Comment by greg-schroeder** [comment]
Any updates here @mananjadhav? Thanks!

**Comment by mananjadhav** [comment]
Apologies for the delay, I was catching up on all issues after past few sick days.

I am working on building the SplitStackView within the existing component. I am also trying to get rid of the `isSmallScreenWidth` check on the responsive navigator. 

https://github.com/Expensify/App/assets/3069065/96d378c1-ab48-4563-8095-e372d22b7f8a

I'll push more updates once I make progress on fixing the issue for all platforms. As you can see it's in a funny shape at the moment. 


Outside of this issue, related to navigation. One of the issues I am working on also requires a way to access the search route from the RHP ([thread](https://expensify.slack.com/archives/C036QM0SLJK/p1718987183771269)).  I am checking if we could get rid of [this](https://github.com/Expensify/App/blob/17e9d095a1603319016529f0eae7eb92c8322fbc/src/libs/Navigation/AppNavigator/createCustomStackNavigator/index.tsx#L65-L79) as well.

**Comment by sakluger** [comment]
Hey @mananjadhav, what's the latest update on this issue?

**Comment by mananjadhav** [comment]
I haven't made good progress on this on the weekend. I'll focus on this PR this week.

**Comment by mananjadhav** [comment]
I'll have the a draft PR to work on Web and iPad by the weekend. Mobile is still a problem but I'll open up a draft PR to confirm the approach and get feedback before getting to fix Mobile app.

**Comment by mountiny** [comment]
@mananjadhav how is the progress looking?

**Comment by sakluger** [comment]
@mananjadhav any updates?

**Comment by mananjadhav** [comment]
I couldn't get this working for the native apps. I am going to put another effort this weekend, else will ask for help/reassign. Apologies for the delay here.

**Comment by sakluger** [comment]
Hey @mananjadhav, did you have any luck with this one?

**Comment by BrtqKr** [comment]
Hey, I'm a developer from SWM I'd like to take over this ticket

**Comment by sakluger** [comment]
Commenting for melvin since we just got an update yesterday.

**Comment by mountiny** [comment]
Makes sense, though could @BrtqKr already be making some progress on this based on the POC branch you have?

**Comment by BrtqKr** [comment]
I'm trying to wire this up with the POC, so I might be able to figure this out in advance, but I'll keep you updated on that during this week.

**Comment by sakluger** [comment]
Hey @BrtqKr any updates?

**Comment by BrtqKr** [comment]
@sakluger, I had to prioritize workspace rules PRs first. I'm done tho, so I'm working on that right now

**Comment by BrtqKr** [comment]
Had some time to experiment with this, this week. With some help from adam we've applied this layout for the tablets using the poc with split stack. I'm trying to research the animations/styling atm, since it's a bit problematic. 

We've got one question about the layout though - do we always want to have a split nav in the portrait layout for the tablets, or maybe just for a landscape? It seemed a bit too narrow in some cases, but there are apps that are approaching it in this manner.

**Comment by mountiny** [comment]
> We've got one question about the layout though - do we always want to have a split nav in the portrait layout for the tablets, or maybe just for a landscape? It seemed a bit too narrow in some cases, but there are apps that are approaching it in this manner.

I don't think this should ever depend on the device and orientation technically; the layout should always be dictated by the viewport size, so if the tablet portrait mode is too narrow, then we would just use narrow mode. Basically the viewport width should be the source of truth for what layout to render

@shawnborton do you agree?

**Comment by shawnborton** [comment]
Totally agree with that Vit, cc @Expensify/design for the gut check too.

**Comment by dannymcclain** [comment]
100% agree with Vit and Shawn.

**Comment by BrtqKr** [comment]
Hey, since most of the things I'm doing atm are just adjustments to the core part, I wanted to give a quick update on how it looks right now. Currently, the navigation is being rendered correctly in the split variant. Regarding the previous question, I mostly meant the tablet breakpoint, but overall it should be quite easy to change afterward. The video below presents a split variant for width >= medium size breakpoint. As you can see, the styling is mostly done, but I'm still trying to properly handle overlays and some more unique places, such as search. I've also cleaned up the breakpoint conditions in multiple places.

The next steps I'm going to take would be: 
1. Finish overlays
2. Experiment with an alternate split stack structure to determine proper state persistence while resizing
3. Test animations
4. Clean up the breakpoints
5. Optimizations if necessary

I'll try to provide you with a full version sometime in the next week 

https://github.com/user-attachments/assets/19d2166d-9720-4f5b-9a96-40a395679f1b

**Comment by mountiny** [comment]
Thanks @BrtqKr! Great to see the progress, please try to post some update every day you work on this, even if there is not much to show visually.

To confirm, are your changes based on the updates from @adamgrzybowski and @WojtekBoman to refactor some of the navigation logic for bottom tabs history persistence? I think you said yes before but in general i assume we want to base this pr on top of it, so we do not have to do more work later

**Comment by BrtqKr** [comment]
sure, sorry for that. I was switching between nav and rules tasks.

Regarding the update for today - I've been working on the custom style interpolator for the modal screens. Everything is related to the animations while transitioning between the report and rhp. I'm making some progress, but I'm somewhere in the middle - there some bugs, which are probably related to the nested stacks, but I'm trying to confirm this

**Comment by adamgrzybowski** [comment]
BTW just to confirm, yes it's based on our changes for the bottom tabs.

**Comment by BrtqKr** [comment]
I've finished the Overlays part, moving to the alternate split stack structure part. Also, we've been testing rendering two screens in the navtive-stack and it will probably require some adjustments, but @adamgrzybowski will be taking care of that.

**Comment by BrtqKr** [comment]
We've been verifying with @adamgrzybowski and @WoLewicki how those changes could be applied if the [PR including native-stack](https://github.com/Expensify/App/pull/37891) gets merged and we've noticed a couple of things. Firstly, the split stack we've been mostly worried about seems to be working as expected, but we had to slightly clean-up the structure. Unfortunately, there are some issues when trying to include slide-in animations for the modals, which would be slightly weird if the original intent was to have a native experience. 

There are some ways to approach this. Here's a list of options we've considered:
1. We could create a modal that would appear without any transition, then just apply the slide for the content using reanimated. This is basically writing our own stack with effects, which is quite ugly and might be annoying to maintain afterward.
2. We could just assume that the modal appears with fade, or something different, but as I stated before, this isn't the native experience we could expect from the modal. 
3. There's an option to keep it in the state from the native stack PR, that is keep is as a full-screen modal. But since we'd like to include horizontal layout in the conditions, this is probably not an option. 
4. We might want to keep some of the modals as a standard stacks, just for the sake of visual flexibility. 

We'd probably suggest option 4, but there might still be some other problems related to the native stack.

**Comment by mountiny** [comment]
Sounds good, lets just see how the options will play out with the bottom tabs + native stack in

**Comment by BrtqKr** [comment]
Had to take care of the extension ticket today. I'll be testing option 4 some time tomorrow or on Wednesday.

**Comment by BrtqKr** [comment]
Still focused on the share extension.

**Comment by BrtqKr** [comment]
Had some time to start applying the solution, but it's still work in progress, not seeing any issues with this approach so far though

**Comment by BrtqKr** [comment]
@mountiny, overall the rendering of the native screens seems to be working in combination with the standard ones, but regarding the modals - we've verified that there might be a risk with rendering increasingly larger parts of the stacks without applying native stack, so before we go further we wanted to confirm if this approach is acceptable.

1. Verify if the RHP as a standard stack with Report split stack written using native stack screens is working as expected. The main issue was the usage of the cards in the native stack and lack of possibilities for customizing modals. As far as we've checked this is unlikely this approach would work, but we'll try to push it slightly further 
2. If there is no chance the first point would work, this might be necessary to rewrite the whole root stack to the previous state and avoid using native screens in there entirely, just to get the customizable modal handling - this is the main reason I prefer to ask before we dig any deeper, because this seems like a very impactful change especially while having native stack PR being worked on in parallel 

We still believe option 4, that is avoiding native stack usage for some of the stacks would be the correct way of handling this scenario, but we wanted to make sure you're aware of where this might lead us.

**Comment by mountiny** [comment]
@BrtqKr, thanks for the update; we are hoping to get the native stack PRs merged soon, so we will be able to see soon. I would say you can try to pull the native stack changes and work based off that, but if you think that its better to wait for those changes to land on main and only then dig deeper, I can put this on hold too

**Comment by sakluger** [comment]
Hey @BrtqKr, what's the latest?

**Comment by mountiny** [comment]
I think we are technically holding for this PR to land as its a big change to the navigation too https://github.com/Expensify/App/pull/49937

**Comment by sakluger** [comment]
The PR linked above is still in review.

**Comment by mountiny** [comment]
Still working on the linked PR

**Comment by mountiny** [comment]
on hold

**Comment by mountiny** [comment]
holding

**Comment by mananjadhav** [comment]
Still held.

**Comment by mountiny** [comment]
On hold

**Comment by sakluger** [comment]
What's going on here @mountiny, is this one still happening?

**Comment by mountiny** [comment]
The native stack pr was merged 🎉 we have another large navigation change under review though so i think it would make sense to put this on hold for it https://github.com/Expensify/App/pull/49539

**Comment by mananjadhav** [comment]
Now that the linked PR is merged, should we unblock this?

**Comment by mountiny** [comment]
@mananjadhav Yes! although I wanted to discuss if you feel up for this one? I think it might be better to give this task to SWM due to its complexity.

**Comment by mountiny** [comment]
Ok cool thanks!

**Comment by mananjadhav** [comment]
Do we have any one assigned on this one to start working then?

**Comment by mountiny** [comment]
SWM is looking into this

**Comment by mountiny** [comment]
Fair, I think we should aim to work on this support for newDot only and we could bring it in when the oldApp is deprecated

**Comment by roryabraham** [comment]
looks like this can come off HOLD

**Comment by roryabraham** [comment]
> I think we should aim to work on this support for newDot only and we could bring it in when the oldApp is deprecated

I'm confused what we are agreeing to here. I agree we only need to support wide layouts in NewDot native and not OldApp, but it should work in HybridApp.

**Comment by mountiny** [comment]
> I'm confused what we are agreeing to here. I agree we only need to support wide layouts in NewDot native and not OldApp, but it should work in HybridApp.

right, yeah we just need to make sure this is handled gracefully in hybridApp

I floated this earlier in the month for SWM https://expensify.slack.com/archives/C08J3UBAAS3/p1741299699865029 however, they are still packed. I think this will have to be handled by agency and SWM is probably the best one as its very complex task

**Comment by mananjadhav** [comment]
Do we have a timeline or a priority set for this one?

**Comment by mountiny** [comment]
Its really on a backburner, no deadline

**Comment by mananjadhav** [comment]
Not overdue, not prioritized.

**Comment by staszekscp** [comment]
Hey everyone! I'd like to pick up this issue 😄 Soon I'll meet with @adamgrzybowski and @WoLewicki to estimate the scope and start digging!

As per:
> I'm confused what we are agreeing to here. I agree we only need to support wide layouts in NewDot native and not OldApp, but it should work in HybridApp.

Yes, theoretically I don't see any major differences that would cause problems with HybridApp implementation - it should be basically the same, but we'll see where I'll end up with it 😄

**Comment by mountiny** [comment]
Nice! Exciting! Let me ask about the android tables, i would think that yes

**Comment by staszekscp** [comment]
Hey! Just a quick update on the work: I was investigating the ability to start with the lowest level of changes (in `react-native-screens`). It required from me implementing `UISplitView`, which I have managed to do, but it still requires much polishing like modifying the shadow tree etc. If we decide to proceed with implementing android, it can will require implementing a similar  thing there, too. In general that would be the cleanest way to approach the wide layout on iPads/tablets. However at the moment I'm not sure if it's worth to continue heading in this direction.

During the work I've understood that the main part of the task will be implementing split view mechanism in the `react-navigation` instead of the `react-native-screens`. I haven't decided yet how to approach it, but I have a couple of ideas. This week I'll try to get some opinions and start working on some kind of POC to see what challenges I'll bump into.

**Comment by mountiny** [comment]
@staszekscp the android tablets are still not clear, I will need to ask about it

**Comment by chuckdries** [comment]
With the rise of folding phones, including the upcoming iPhone Duo, this is going to be something more and more users actually notice. I'm reopening so I can work on https://github.com/Expensify/App/pull/101015

**Comment by dubielzyk-expensify** [comment]
cc @Expensify/design

**Comment by trjExpensify** [comment]
Haha, this is a mega old issue from 2024 with 200+ comments. Can't we parent https://github.com/Expensify/App/pull/101015 to a fresh issue with a revised and up-to-date plan?
